src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs

```diff
@@ -911,24 +867,38 @@ private void InitializeFSM()
                     case OldestChangedBuffer.OldestChanged oldestChanged when e.StateData is YoungerData youngerData:
                     {
                         _oldestChangedReceived = true;
-                        if (oldestChanged.Oldest != null && oldestChanged.Oldest.Equals(_selfUniqueAddress))
+                        if (oldestChanged.NewOldest != null && oldestChanged.NewOldest.Equals(_selfUniqueAddress))
                         {
                             Log.Info("Younger observed OldestChanged: [{0} -> myself]", youngerData.Oldest.Head()?.Address);
-
                             if (youngerData.Oldest.All(m => _removed.ContainsKey(m)))
                             {
                                 return TryGotoOldest();
                             }
 
-                            Peer(youngerData.Oldest.Head().Address).Tell(HandOverToMe.Instance);
-                            return GoTo(ClusterSingletonState.BecomingOldest).Using(new BecomingOldestData(youngerData.Oldest));
+                            // explicitly re-order the list to make sure that the oldest, as indicated to us by the OldestChangedBuffer,
+                            //  is the first element - resolves bug https://github.com/akkadotnet/akka.net/issues/6973
+                            var newOldestState = oldestChanged.PreviousOldest switch
```

Comment:
Re-arrange the state we're going to retain so we make sure the `PreviousOldest` node, as reported by the `OldestChangedBuffer`, appears at the front of the list. This ensures that the `BecomingOldest` node is always performing the hand-off with the correct party in the event of a rolling update.